File: SceneModel/Sex.cs
namespace SceneModel
{
    public enum Sex
    {
        Any = 0,
        Male = 1,
        Female = 2,
        Both = 3
    }

    public static class SexExtension
    {
        private static readonly string[] map = new string[4] { "P", "M", "F", "A" };
        public static string Abbr(this Sex sex)
        {
            return map[(int)sex];
        }
    }
}

File: F4SEPlugin/Sex.cpp
#include "Sex.h"

const Data::Sex Data::Sex::Any(0);
const Data::Sex Data::Sex::Male(1);
const Data::Sex Data::Sex::Female(2);

std::string_view map[3]{"P", "M", "F"};
Data::Sex::operator std::string_view() const
{
	return map[value];
}

